Creating an RPM package for your software (#27)
oracle/ol-sample-scripts: creating-rpm-packages/creating-rpm-packages.md

# Creating an RPM package for your software

[RPM](https://en.wikipedia.org/wiki/RPM_Package_Manager) is both a

> package manager and a file format (.rpm). It has been used by many linux

Reviewer: ```suggestion
package manager and a file format (.rpm). It has been used by many Linux
```

distributions, including Oracle Linux, to ease the distribution,
installation and maintenance of software. In this document we\'ll
explain how to create an rpm using a simple example.

## Why create an RPM

Packaging software with RPM provides many advantages over delivering

> software via an simple tarball archive file. The main advantages are:

Reviewer: ```suggestion
software via an simple tarball. The main advantages are:
```

    by using package management tools such as Yum
- Ease of distribution
  - RPM files are self describing, they can contain information such
    as a description, installation instruction, list of files,
    dependencies version, etc.  \
    This helps users quickly understand what the software package is
    about, what version it is, what are its dependencies, how to
    install, etc.
  - RPMs can be added to repositories such as Yum (or in a software
    source via the OS Management service in [Oracle Cloud
    Infrastructure](https://www.oracle.com/cloud/) that allows users
    to find and install your software.
- Ease of integration
  - Linux distributions such as Oracle Linux use RPM to deliver
    software packages and keep track of what\'s installed in a
    central database. This avoids re-installation of an already
    installed software package and can also be used to determine
    which package installed files belong to.
- Ease of authentication
  - RPM can be signed using GPG to enable the user to verify the
    origin of a package.

## Overview

The main steps for packaging software are:

- Prepare your system to build an rpm

> - Get our software to build and install on your system

Reviewer: ```suggestion
- Verify that the software to be packaged builds and installs on your system
```
